### PR TITLE
Export from redux store

### PR DESCRIPTION
**Pull request by akofink**

### CODE REVIEW

File: src/store/Reducers/SystemStore.js [hunk 1]
```diff
@@ -79,6 +84,40 @@ export const systemsToInventoryEntities = (systems, entities) =>
...
+
+const csvFromState = (state) => {
+    if (state.rows) {
+        const CELL_DELIMITER = ',';
+        let csvRows = [state.columns.map((column) => column.title).join(CELL_DELIMITER)];
+        csvRows = csvRows.concat(state.rows.map((row) => {
+            return state.columns.map((column) => {
+                let cell = get(row, column.key);
+                if (typeof(cell) === 'object') {
+                    cell = get(row, column.key + 'Text');
```
Comment (akofink) [inline src/store/Reducers/SystemStore.js]: So, to make this function more reusable, I've added this convention here that if a column is an object, try to get the column key with 'Text' appended. i.e. `facts.compliance.compliant` is a react component object, so we look for the plain-text version at `facts.compliance.compliantText`. This is fairly trivial to add to a reducer, as I've done on line 78.

File: src/store/Reducers/SystemStore.js [hunk 2]
```diff
@@ -69,16 +77,54 @@ export const systemsToInventoryEntities = (systems, entities) =>
...
+
+const csvFromState = (state) => {
+    if (state.rows) {
+        const CELL_DELIMITER = ',';
+        let csvRows = [state.columns.map((column) => column.title).join(CELL_DELIMITER)];
+        csvRows = csvRows.concat(state.rows.map((row) => {
+            return state.columns.map((column) => {
+                let cell = get(row, column.key);
+                if (typeof(cell) === 'object') {
+                    cell = get(row, column.key + '_text');
```
Comment (akofink) [inline src/store/Reducers/SystemStore.js]: So, to make this function more reusable, I've added this convention here that if a column is an object, try to get the column key with '_text' appended. i.e. `facts.compliance.compliant` is a react component object, so we look for the plain-text version at `facts.compliance.compliant_text`. This is fairly trivial to add to a reducer, as I've done on line 84-86.

File: src/store/Reducers/SystemStore.js [hunk 3]
```diff
@@ -69,16 +77,54 @@ export const systemsToInventoryEntities = (systems, entities) =>
...
                             }
                         }}>{system.rules_failed}</Link>,
+                        rules_failed_text: system.rules_failed,
                         compliance_score: complianceScore(system),
+                        compliance_score_text: complianceScoreString(system),
                         compliant: compliantIcon(system),
-                        last_scanned: <FormattedRelative value={Date.parse(system.last_scanned)} />
+                        compliant_text: system.compliant.toString(),
+                        last_scanned: <FormattedRelative value={Date.parse(system.last_scanned)} />,
+                        last_scanned_text: system.last_scanned
```
Comment (akofink) [inline src/store/Reducers/SystemStore.js]: I know the original bug was that the export didn't look exactly like the html table; however, we definitely want parts of it to be different, like here: a relative date in a csv is pretty much useless IMO. Best to use the full date/time format with timezone to be clear.

File: src/store/Reducers/SystemStore.js [hunk 4]
```diff
@@ -69,16 +77,54 @@ export const systemsToInventoryEntities = (systems, entities) =>
...
                     }
                 }
                 /* eslint-enable camelcase */
             };
         }
     ).filter(value => value !== undefined);
 
+const linkAndDownload = (data, filename) => {
```
Comment (dLobatog) [inline src/store/Reducers/SystemStore.js]: Ideally all of these methods would live in `src/Utilities/CsvExport` or somewhere else in `src/Utilities`

File: src/store/Reducers/SystemStore.js [hunk 5]
```diff
@@ -69,16 +77,54 @@ export const systemsToInventoryEntities = (systems, entities) =>
...
+};
+
+const csvFromState = (state) => {
+    if (state.rows) {
+        const CELL_DELIMITER = ',';
+        let csvRows = [state.columns.map((column) => column.title).join(CELL_DELIMITER)];
+        csvRows = csvRows.concat(state.rows.map((row) => {
+            return state.columns.map((column) => {
+                let cell = get(row, column.key);
+                if (typeof(cell) === 'object') {
```
[the remark this answers is not in the document]
Comment (akofink) [reply src/store/Reducers/SystemStore.js]: It looks like it's pretty standard practice to wrap those fields in quotes: `foo, bar, "PCI for RHEL 7, Standard for RHEL 7", baz`. When I open such a csv in libreoffice, it works with default import settings. What do you think? I'd prefer this to changing the field data itself.

[the remark this answers is not in the document]
Comment (dLobatog) [reply src/store/Reducers/SystemStore.js]: That sounds good, anything that makes it clear that's a column would work

**Comment by akofink** [comment]
Updated - also, we need to handle JSON too :crying_cat_face:

**Pull request merged**